### PR TITLE
feat: Add Trunk, Next.js, Nuxt Frontend Guides

### PR DESCRIPTION
**Pull request by simonhyll**
Technically this might've spiraled out of control into more of a "framework configuration" PR but I started it as a PR for mobile configuration so damn it that's what the branch is gonna be for!

Currently looking for some input on whether this is a desired format. I kinda think these sort of framework configurations fit better into a section about project creation but I'm not sure.

Currently only the Next.js article is anywhere close to finished, the rest have yet to be written.

### CODE REVIEW

**Comment by lorenzolewis** [comment]
Note to self from Discord https://discord.com/channels/616186924390023171/662624589037436928/1141394287901290657:

- Nice checklist CSS? Is that something that's exposed in Starlight that I can steal?
- Think through page structure and where this would slot in the best in the page hierarchy. 

@FabianLars if you have any feedback on this (level of detail and technical content in particular) then would love to get your input as well.

**Comment by lorenzolewis** [comment]
Thinking of the flow, this would make the most sense for a person to read right after https://beta.tauri.app/2/guide/create/, right? I'd imagine this would be useful to surface very visibly and maybe put directly under the "Quick Start" sidebar category? And then shift the stuff in `guide/develop` to something like `guide/framework/index`?

I started thinking this weekend about an end to end DX/QA pass on 2.0 and I might ask the dev team if they'd be up to a challenge to see how much of that could be automated into the Tauri CLI when the audit kicks off (which could mean one less page in the sidebar 🥳 )

**Comment by lorenzolewis** [comment]
How accurate are all of these guides? If the next.js one is the only one near finished could we wrap that one up and then use the `<Stub />` component (and include any notes you have) on the other pages and ask the community to contribute?

**Comment by FabianLars** [comment]
> @FabianLars if you have any feedback on this (level of detail and technical content in particular) then would love to get your input as well.

So far it looks exactly like what i've had in mind (minus the everything on one page part but that may not work nicely anyway)

> I started thinking this weekend about an end to end DX/QA pass on 2.0 and I might ask the dev team if they'd be up to a challenge to see how much of that could be automated into the Tauri CLI when the audit kicks off (which could mean one less page in the sidebar 🥳 )

Stop :joy: srsly tho, this is not something i'd like to outsource/upstream into the cli especially since some frontends like to change small things all the time and it would feed into the "tauri supports these frameworks, but no more" even more than just having guides imo.

**Comment by lorenzolewis** [comment]
@FabianLars that was more about the prerequisites (Xcode, Rust, etc., similar to how CTA does it now), not the per-framework setup. But yes, point taken 😉

**Comment by FabianLars** [comment]
Funnily the first draft of the message was about that. But yeah, it still applies to that too, realistically we could only autoinstall the deps on Windows if there's no visual studio installation and on Linux we'd have to cover 100 diff distros. No idea how reliably we can work with Android studio. Xcode I think is possible? It should be possible to install store apps from the cli  🤔
But yeah, that's something we talked about many times and for v1 chip even worked on _something_

**Review by lorenzolewis — CHANGES_REQUESTED** [review]
Hope you don't mind but did a bit of re-work on this one to get it closer to the finish line:

- Moved intro section that was in `guide/develop.mdx` into `guide/framework/index.mdx` and referenced to that in `guide/frontend/index.mdx`
- Changed recommended CTA setup to Vanilla instead of Solid.js in `guide/create/index.mdx`
- Rephrased some things to try to be a bit more succinct. 

A few points on feedback:

- You said that currently only the Next.js guide is complete. Can we remove the content from all the others and instead use a `<Stub />` component (with any relevant notes inside that should be incorporated)?
- What do you think about moving any mobile-specific things into their own heading? That way it's very clear what's required for a desktop app vs. what's an extra step that's needed for mobile

I think we could easily look at making a few components/fragments for the sake of DRY across the guides, but let's get a single guide out and then see what we can refactor into components as we add more in.

**Comment by simonhyll** [comment]
So much activity while I've been at work 👀

The Next.js and Nuxt ones are more or less good as they are afaik and could be merged. Some of the others (mainly Rust based ones) also have some useful info in them, so not sure we want to remove them. Maybe add a stub thingie on top with something like "Needs updates"? Idk just dont want that "hard earned knowledge" to get lost. I can sit and update them a bit during the weekend possibly when I'm not studying for certificates.

I liked the frontend agnostic separate page. Sad panda is sad. But I also have no good argument against your approach of putting it straight on the page.

I think it being moved to where it is makes more sense indeed, right after creating a project.

Really like the big purple box saying the frameworks arent officially supported and unlisted stuff might be good to go as well.

I dont think we should make mobile its own section since the setup works for both, it makes little sense to give people a config that works 100% for desktop but only 70% for mobile, I'd rather everyone just configure their project to be 100% compatible for both straight away, less risk of people making mistakes while developing in case some feature of some framework doesnt work when they later make their project mobile capable and suddenly have to rethink a bunch of their setup.

File: src/content/docs/2/guide/frontend/sveltekit.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/sveltekit.mdx:1]: I'm going to open the age-old question: Do we say Svelte, SvelteKit, Svelte + Vite, etc.?

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/trunk.mdx]: Alright, let's open the can of worms on this one 🪱

1. Is this fork something that can be incorporated upstream into trunk?
2. If so, is there a PR for it we can track?
3. Is this only required for 2.0/mobile? Or is this also required for 1.x?

We can't point to forks in docs because 1) we (or the individual who maintains that fork) are on the hook to maintain it and 2) We should solve this at the source (i.e. trunk or Tauri) if at all possible and get it merged in

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/trunk.mdx]: Ditto

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/trunk.mdx]: Ditto

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/trunk.mdx]: `index.html` is the default here, right? Does that result in the same behaviour as `./index.html`? If so, then we don't need to list it.

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/trunk.mdx]: Defaults to false, I don't think we need to list this

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/trunk.mdx]: Is there a reason we're specifying this instead of leaving it at the default `8080`?

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/trunk.mdx]: Same point on defaults

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/trunk.mdx]: I couldn't find this option in the trunk docs 🤔

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/trunk.mdx]: Is this applicable for Trunk?

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/nuxt.mdx]:
```suggestion
```
We should leave this up to the reader. We as Tauri cover REQUIREMENTS, not preferences 💜

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/nuxt.mdx]: Is this required to get it up and running with Tauri?

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/nuxt.mdx]: Are these actually required anywhere to get a Tauri project working?

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/nuxt.mdx:97]: I couldn't find a lot of details in the vite docs: https://vitejs.dev/config/server-options.html#server-hmr.

- I'm assuming all of these are required for mobile development?
- Does `host` need to be explicitly set or does it inherit it from `server.host`?
- Is there a reason `port` is a different value than the default `3000`?

File: src/content/docs/2/guide/frontend/nextjs.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/nextjs.mdx]:
```suggestion
```

This defaults to true since 13.4

File: src/content/docs/2/guide/frontend/nextjs.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/nextjs.mdx]:
```suggestion
```
This is true as of v13 https://nextjs.org/docs/pages/building-your-application/upgrading/version-13#v13-summary

**Review by lorenzolewis — COMMENTED** [review]
Did a chunk of stuff on this one and my brain is fried from parsing through framework config documentation 🤯 I turned some pages into stubs, most of them didn't have any edits and were just copy-paste from the Next.js page it looks like, so no info lost there. Did a few tweaks for some phrasings, linked out in more places, and added versions for what the guide was written for.

I put some feedback questions but I think I may have more. But tbh that's all I can handle this evening 😅 Major props to you for keeping all of these intricacies in your head.

File: src/content/docs/2/guide/frontend/sveltekit.mdx
Comment (simonhyll) [reply src/content/docs/2/guide/frontend/sveltekit.mdx:1]: I want to say Svelte + Vite but dont trust me on it

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (simonhyll) [reply src/content/docs/2/guide/frontend/trunk.mdx]: Can be? Probably. Will be? No idea
@amrbashir thoughts?

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (simonhyll) [reply src/content/docs/2/guide/frontend/trunk.mdx]: Pikachu

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (simonhyll) [reply src/content/docs/2/guide/frontend/trunk.mdx]: Patrick Swayze

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (simonhyll) [reply src/content/docs/2/guide/frontend/trunk.mdx]: I think you may be right but will need to confirm to be sure

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (simonhyll) [reply src/content/docs/2/guide/frontend/nuxt.mdx]: No but I'd argue showing them off will help people and since the comment says it's optional I think it'd help more people than it might potentially "harm". Remove it if you wanna keep things minimal

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (simonhyll) [reply src/content/docs/2/guide/frontend/nuxt.mdx]: That part I took from the existing Vite docs in the next branch, haven't used this part myself but supposedly it helps.

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (simonhyll) [reply src/content/docs/2/guide/frontend/nuxt.mdx:97]: - it's to fix hot reloading for mobile, yes
- in vite it has to be explicit, i **think** it can be skipped for nuxt, unverified
- yes, because it's 2 different servers, 3000 is the http server while 5183 is the websocket server for hmr (dont ask me why that specific port)

**Comment by lorenzolewis** [comment]
I was also thinking about how we show different configurations through different framework versions and figured that would be a good use of tabs instead of inline comments in code blocks when we hit that scenario.

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (simonhyll) [reply src/content/docs/2/guide/frontend/trunk.mdx]: Convention/familiarity I suppose, we have a lot of templates that default to it. Could probably br removed

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (simonhyll) [reply src/content/docs/2/guide/frontend/trunk.mdx]: We can remove it

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (simonhyll) [reply src/content/docs/2/guide/frontend/trunk.mdx]: Because it's an Amr special
<https://github.com/amrbashir/trunk/commit/0eea992979242f3812140b334e33229685fdc89b>

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (simonhyll) [reply src/content/docs/2/guide/frontend/trunk.mdx]: Generally applicable everywhere, doesnt hurt to mention it. Never know what people do otherwise

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (simonhyll) [reply src/content/docs/2/guide/frontend/nuxt.mdx]: Could fall under the category of best practises, telemetry should be opt-in and Tauri helps keep our users safe and protect their privacy

**Comment by simonhyll** [comment]
So a new tab for each version of the framework?
I'm on board with that 🙂

**Comment by lorenzolewis** [comment]
Yeah, or at least each version that requires a different configuration setup.

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (amrbashir) [reply src/content/docs/2/guide/frontend/trunk.mdx]: 1. I have already opened the PRs before deciding on a fork because it is taking a while to get them merged.
2. https://github.com/thedodd/trunk/pull/494 and https://github.com/thedodd/trunk/pull/500
3. Yeah only for mobile, if using only desktop whether it is 1.0 or 2.0, you can just use the original `trunk`

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (amrbashir) [reply src/content/docs/2/guide/frontend/nuxt.mdx]: I actually cleaned this up a bit in CTA https://github.com/tauri-apps/create-tauri-app/commit/b586eef0c5d5462dbcc84aa1704dd41b03e0e4c2, the defaults of vite build options are fine for tauri.

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (amrbashir) [reply src/content/docs/2/guide/frontend/nuxt.mdx]: Not required but convenient to show, otherwise people won't know about it

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [reply src/content/docs/2/guide/frontend/trunk.mdx]: Lol, I love Amr special things 💜. I'm going to shift this up to the callout above then

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [reply src/content/docs/2/guide/frontend/trunk.mdx]: Alright, let me ask a question in the PR body on this to discuss then, have an idea I want feedback on.

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/trunk.mdx]: Is it common in other Rust projects to document `bininstall` in their steps? I don't see it mentioned in the Trunk docs. By putting this here could it cause potential issues or questions from a reader who tries this and doesn't know what they're doing?

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [reply src/content/docs/2/guide/frontend/trunk.mdx]: Adjusted the phrasing on this. Let me know if you all are comfortable with how it's shown now.

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [reply src/content/docs/2/guide/frontend/trunk.mdx]: Also impacts https://github.com/tauri-apps/tauri-docs/pull/1405/files#diff-20c8897908cc041a1fc1be23670a08a21c4c779cd020a96e6db7622919a83bc8R13

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/nuxt.mdx:73]:
```suggestion
	ssr: false,
```

How about we try this where we move it under the checklist as optional? I'm not sure I'd really phrase that as a best practice, it's very project-dependent depending on the application and the company/team/person developing it.

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (lorenzolewis) [reply src/content/docs/2/guide/frontend/nuxt.mdx]: I put a note under the checklist to cover this. I agree that it could be a point on safety and privacy, but I think that's a user-preference dependent on the company/project/person and not necessarily a best practice 🤔

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (lorenzolewis) [reply src/content/docs/2/guide/frontend/nuxt.mdx]: What's "convenient" mean in this case? 😅 By not having it would it cause issues for a dev when using Tauri? If that's the case then awesome to keep it in.

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (lorenzolewis) [reply src/content/docs/2/guide/frontend/nuxt.mdx]: Let's remove this then to match CTA.

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (lorenzolewis) [reply src/content/docs/2/guide/frontend/nuxt.mdx:97]: Cool, will comment on ports in a separate comment.

**Comment by lorenzolewis** [comment]
Alright, this one is sooooooo close! 🥳 Just a few more details:

- Confirm https://github.com/tauri-apps/tauri-docs/pull/1405#discussion_r1313345345
- Does the adjustment for the trunk fork stuff look okay?
- https://github.com/tauri-apps/tauri-docs/pull/1405#discussion_r1313830153 If this can lead to broken builds/dev by not being there then let's keep it
- https://github.com/tauri-apps/tauri-docs/pull/1405#discussion_r1313827436 Advice from you experts on bininstall and if it's something we should mention or if we leave it to the reader to pull in if they're aware of it (we could also add a note to the top to point them to the documentation instead of using tabs here as an alternative)
- Ports: Do we want to hard-code all the ports as `1420` in the guides to match CTA? This might be one of the "not required but will improve support" things I'm okay with. If so, what port do we want do we want to use for cases like this to also be consistent: https://github.com/tauri-apps/tauri-docs/blob/42807bb95439a3ba4f853f4646563ff1014eca91/src/content/docs/2/guide/frontend/nuxt.mdx#L91 
- Renamed the svelte doc from `sveltekit.mdx` to `svelte.mdx` to give us some flexibility to change the title to whatever we want in the future without affecting the slug (can handle that convo later on).
- https://github.com/tauri-apps/tauri-docs/pull/1405#discussion_r1313355325 fun telemetry convo 🙃

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (amrbashir) [reply src/content/docs/2/guide/frontend/nuxt.mdx]: Doesn't cause an issue but the info would be kinda lost and can't search for it since this is the only place that shows it?

Would be obsolete when I cleanup https://github.com/tauri-apps/tauri/blob/dev/tooling/cli/ENVIRONMENT_VARIABLES.md and make the naming consistent in v2, then we can have a page for these env vars similar to this https://doc.rust-lang.org/cargo/reference/environment-variables.html

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (amrbashir) [reply src/content/docs/2/guide/frontend/trunk.mdx]: `cargo binstall` is probably not supported for my fork anyways, you should use `cargo install --git`.

Comment (simonhyll) [reply src/content/docs/2/guide/frontend/trunk.mdx]: Yea binstall I don't think works either since it relies on finding a prebuilt version in Github releases. I primarily added it as a note to myself to remember binstall for other projects that support it

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (simonhyll) [inline src/content/docs/2/guide/frontend/trunk.mdx]: Maybe reword this? Just like "possible with *the official version of* Trunk"
```suggestion
Tauri mobile development is not currently possible with the official version of Trunk. If developing for mobile you will need to use https://github.com/amrbashir/trunk until https://github.com/thedodd/trunk/pull/494 and https://github.com/thedodd/trunk/pull/500 are merged and released.
```

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (lorenzolewis) [reply src/content/docs/2/guide/frontend/nuxt.mdx]: Ahhh I never knew that page existed! I think a page like that would be better to keep in the `tauri` repo so that it's more easily kept accurate and then we can link to it from here. Would that work? https://github.com/tauri-apps/tauri/blob/next/tooling/cli/ENVIRONMENT_VARIABLES.md

If that's okay then we can adapt the docs here to highlight that.

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [reply src/content/docs/2/guide/frontend/trunk.mdx]: Cool, just updated in the callout and also removed that section. Thanks!

**Comment by lorenzolewis** [comment]
Alright, last thing is on ports: do we want to put `1420` on all the guides?

**Comment by amrbashir** [comment]
I'd recommend sticking with the framework's defaults, so users won't have to configure the framework server.

**Comment by lorenzolewis** [comment]
Cool, only one that needed updated was Trunk based off of https://github.com/thedodd/trunk/blob/7a27876ad1e38684ceb78239023a421dc1f46e84/Trunk.toml#L27

@amrbashir do you think it would make sense to update CTA eventually to make it all match the guide as well?

@simonhyll how does this all look to you? Are you alright to merge as-is in the current state (and if so then feel free to hit that big green merge button)?

---

On a complete side note: One idea I'd love to explore in the future is some sort of adaptor model for us to get as close to zero-config as possible with frameworks. I know we explored this a bit with SvelteKit. If we do that then we can in theory have CTA driven off of that, just point to an adaptor in docs, and then for manual setups just point to the code where the adaptor is applying the config. But just food for thought right now 🥘

**Review by lorenzolewis — APPROVED** [review]

File: src/content/docs/2/guide/frontend/index.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/index.mdx]:
```suggestion
---
title: Frontend Configuration
i18nReady: true
---
```

File: src/content/docs/2/guide/frontend/nextjs.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/nextjs.mdx]:
```suggestion
---
title: Next.js
i18nReady: true
---
```

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/nuxt.mdx]:
```suggestion
---
title: Nuxt
i18nReady: true
---
```

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (lorenzolewis) [inline src/content/docs/2/guide/frontend/trunk.mdx]:
```suggestion
---
title: Trunk
i18nReady: true
---
```

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (amrbashir) [reply src/content/docs/2/guide/frontend/nuxt.mdx]: We can't link to it, we need to index it and make it searchable, just need to polish it up a bit on tauri's side and then it can be copied to the docs on new releases.

Comment (lorenzolewis) [reply src/content/docs/2/guide/frontend/nuxt.mdx]: Ahhhhh... yeah good point, let me stub that page then and we can link to at least the stub in this PR.

**Comment by amrbashir** [comment]
> do you think it would make sense to update CTA eventually to make it all match the guide as well?

Do you mean the ports? probably no, for CTA, I think it is better to add fixed ports, to stay clear from other projects potentially occupying the port.

File: src/content/docs/2/guide/frontend/nuxt.mdx
Comment (lorenzolewis) [reply src/content/docs/2/guide/frontend/nuxt.mdx]: There, added that in now and marked it as a stub so we know it's a TODO for the docs.

**Review by simonhyll — COMMENTED** [review]
Looks good to me 😄

File: src/content/docs/2/guide/frontend/trunk.mdx
Comment (simonhyll) [inline src/content/docs/2/guide/frontend/trunk.mdx:19]:
```suggestion
- Use `address = "0.0.0.0"` so that the webserver is available on the network for mobile development.
- Enable `withGlobalTauri` to ensure that Tauri APIs are available in the `window.__TAURI__` variable and can be imported using `wasm-bindgen`.
```

**Pull request merged**
